cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -2013,8 +2013,13 @@ Result<std::shared_ptr<ChunkedArray>> TakeCA(const ChunkedArray& values,
     // TODO Case 3: If indices are sorted, can slice them and call Array Take
 
     // Case 4: Else, concatenate chunks and call Array Take
-    ARROW_ASSIGN_OR_RAISE(current_chunk,
-                          Concatenate(values.chunks(), ctx->memory_pool()));
+    if (values.chunks().empty()) {
+      ARROW_ASSIGN_OR_RAISE(current_chunk, MakeArrayOfNull(values.type(), /*length=*/0,
+                                                           ctx->memory_pool()));
+    } else {
+      ARROW_ASSIGN_OR_RAISE(current_chunk,
+                            Concatenate(values.chunks(), ctx->memory_pool()));
+    }
```

Comment:
Do we need to apply this same fix for TakeCC below?

Note: TakeCC might not be in the GroupByNode path so this could maybe be done as a separate JIRA.

Reply:
No, because if we're in the loop body, then we're calling Concatenate on the result of TakeCA, and TakeCA always returns one chunk. However, I've added more tests for this case.